core/domain/suggestion_services.py

```diff
@@ -1798,27 +1807,34 @@ def update_question_suggestion(
         Suggestion|None. The corresponding suggestion, or None if no suggestion
         is found.
     """
-    suggestion = get_suggestion_by_id(suggestion_id)
+    # Here we use cast because we are narrowing down the type from
```

Comment:
Ditto here, I feel this isn't an appropriate use of cast() since it relies on unchecked assumptions.

Reply:
Done